orchagent/muxorch.cpp

```diff
@@ -983,6 +1075,91 @@ sai_object_id_t MuxOrch::getNextHopTunnelId(std::string tunnelKey, IpAddress& ip
     return it->second.nh_id;
 }
 
+/**
+ * @brief updates the given route to point to a single active NH or tunnel
+ * @param pfx IpPrefix of route to update
+ * @param remove bool only true when route is getting removed
+ */
+void MuxOrch::updateRoute(const IpPrefix &pfx, bool remove)
+{
+    NextHopGroupKey nhg_key;
+    NextHopGroupEntry nhg_entry;
+
+    if (remove)
+    {
+        mux_multi_nh_route_tb.erase(pfx);
+        return;
+    }
+
+    /* get nexthop group key from syncd */
+    nhg_key = gRouteOrch->getSyncdRouteNhgKey(gVirtualRouterId, pfx);
+
+    /* check for cross-mux neighbors */
+    if (nhg_key.getSize() > 1)
+    {
+        std::set<NextHopKey> nextHops;
+        sai_object_id_t next_hop_id;
+        sai_status_t status;
+        bool active_found = false;
+
+        /* get nexthops from nexthop group */
+        nextHops = nhg_key.getNextHops();
+
+        auto it = mux_multi_nh_route_tb.find(pfx);
+        if (it != mux_multi_nh_route_tb.end())
+        {
+            MuxCable* cable = findMuxCableInSubnet(it->second.ip_address);
+            if (cable == nullptr || cable->isActive())
+            {
+                SWSS_LOG_NOTICE("Route %s pointing to active neighbor %s",
+                    pfx.to_string().c_str(), it->second.to_string().c_str());
+                return;
+            }
+        }
+
+        SWSS_LOG_NOTICE("Updating route %s pointing to Mux nexthops %s",
+                    pfx.to_string().c_str(), nhg_key.to_string().c_str());
+
+        for (auto it = nextHops.begin(); it != nextHops.end(); it++)
+        {
+            NextHopKey nexthop = *it;
+            MuxCable* cable = findMuxCableInSubnet(nexthop.ip_address);
+            if (cable == nullptr || (cable->isActive() && !active_found))
```

Comment:
same as above comment - if the nexthop IP isn't a mux cable IP, it seems safer to assume it's standby

Reply:
same as above